provider/internal/user_api_token/resource.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var _ resource.Resource = &Resource{}
+var _ resource.ResourceWithImportState = &Resource{}
+
+func NewResource() resource.Resource {
+	return &Resource{}
+}
+
+type Resource struct {
+	client *cdoClient.Client
+}
+
+type ApiTokenResourceModel struct {
+	Username types.String `tfsdk:"username"`
+	ApiToken types.String `tfsdk:"api_token"`
+}
+
+func (r *Resource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*cdoClient.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *cdoClient.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *Resource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_api_token"
+}
+
+func (r *Resource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides a resource to generate an API token for an API-only user. This allows an API-only user's token to be created and refreshed on CDO.",
+
+		Attributes: map[string]schema.Attribute{
+			"username": schema.StringAttribute{
+				MarkdownDescription: "The username for which the API token should be generated. This username must be of an [API only user](https://www.cisco.com/c/en/us/td/docs/security/cdo/managing-ftd-with-cdo/managing-ftd-with-cisco-defense-orchestrator/basics-of-cisco-defense-orchestrator.html?bookSearch=true#Cisco_Task.dita_d5ae397b-5aa5-4de0-82c1-a4aff63c5ba1).",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"api_token": schema.StringAttribute{
+				MarkdownDescription: "The API token for the user. This API token has no expiry; to re-generate it, delete the resource and recreate it.",
+				Computed:            true,
+				Sensitive:           true,
+			},
+		},
+	}
+}
+
+func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, res *resource.CreateResponse) {
+	tflog.Trace(ctx, "create API token resource for user")
+
+	// 1. read terraform plan data into model
+	var planData ApiTokenResourceModel
+	res.Diagnostics.Append(req.Plan.Get(ctx, &planData)...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	// 2. create resource & fill model data
+	generateApiTokenInp := user.NewGenerateApiTokenInput(planData.Username.ValueString())
```

Comment:
I disagree strongly. We need to call things what they actually mean, or contributors who work on CDO will not be able to understanding mappings between what actually happens and what the terraform provider does.

Reply:
I feel like clicking into it and see the type alias is pretty much the same, but yea, if it reduce learning curve for other developers, go for it. In the future, we will be developing the go client and if we have external contributor they will be only working under `/provider` directory right?

Reply:
I'm hoping eventually (this will probably never happen) we'll auto-generate a go client from our public API. :)

Reply:
But until then, the probability of a non-CDO engineer developing any code here is slim to non-existent because the client uses our hard to understand APIs.